api/views/enrollment_test.py

```diff
@@ -16,4 +22,21 @@ class EnrollmentTestViewSet(ReadOnlyModelViewSet[EnrollmentTest]):
 
 class EnrollmentTestResultViewSet(CreateModelMixin, GenericViewSet[EnrollmentTestResult]):
     queryset = EnrollmentTestResult.objects.all()
-    serializer_class = EnrollmentTestResultCreateSerializer
+
+    def get_serializer_class(self) -> type[BaseSerializer[EnrollmentTestResult]]:
+        if self.action == "get_level":
+            return EnrollmentTestResultLevelSerializer
+        return EnrollmentTestResultCreateSerializer
+
+    @action(detail=False, methods=["post"])
+    def get_level(self, request: Request) -> Response:
+        """Calculates level of language based on number of correct answers."""
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            return Response({"resulting_level": serializer.data["resulting_level"]})
+        except NotImplementedError:
+            raise ValidationError(
+                f"Enrollment test with {len(serializer.validated_data['answers'])} "
```

Comment:
I think here you could use serializer.data to make it shorter

Reply:
Also, this excepting here doesn't look good, I think it should be excepted in serializer's validate method

Reply:
Oh, I actually should have deleted the whole `try...except` bit and just leave `return` statement. The block was there before I implemented `.validate` in the serializer. Removed.